pkg/tbtc/maintainer/chain.go

```diff
@@ -2,12 +2,38 @@ package maintainer
 
 import (
 	"github.com/keep-network/keep-core/pkg/bitcoin"
+	"github.com/keep-network/keep-core/pkg/chain"
 )
 
 // BitcoinDifficultyChain is an interface that provides the ability to
 // communicate with the Bitcoin difficulty on-chain contract.
 type BitcoinDifficultyChain interface {
-	// Retarget adds a new epoch to the Bitcoin difficulty relay by providing
-	// a proof of the difficulty before and after the retarget.
-	Retarget(headers []bitcoin.BlockHeader) error
+	// Ready checks whether the relay is active (i.e. genesis has been performed).
+	// Note that if the relay is used by querying the current and previous epoch
+	// difficulty, at least one retarget needs to be provided after genesis;
+	// otherwise the prevEpochDifficulty will be uninitialised and zero.
+	Ready() (bool, error)
+
+	// AuthorizationRequired checks whether the relay requires the address
+	// submitting a retarget to be authorised in advance by governance.
+	AuthorizationRequired() (bool, error)
+
+	// IsAuthorized checks whether the given address has been authorised by
+	// governance to submit a retarget.
+	IsAuthorized(address chain.Address) (bool, error)
+
+	// Signing returns the signing associated with the chain.
+	Signing() chain.Signing
```

Comment:
I'm not sure we should define this function in `BitcoinDifficultyChain` interface. It is already exposed in the `tbtc.Chain` interface:
 https://github.com/keep-network/keep-core/blob/e49fef51fdb56e470a49577cca339d05c9ab0f15/pkg/tbtc/chain.go#L119

Reply:
If we do https://github.com/keep-network/keep-core/pull/3394/files#r1038165482, I think it should. The `BitcoinDifficultyChain/LightRelayChain` must not depend on the `tbtc.Chain` and should declare its own expectations. That will lead to the situation where both `tbtc` and `maintainer` packages will declare their own `Chain` interfaces both focused on the tBTC sub-domains specific for those packages. For example,
- `tbtc.Chain` will declare `Bridge` functions essential for operations performed by the wallets itself, like getting revealed deposits and so on
- `maintainer.Chain` (apart from light relay functions) will declare `Bridge` functions responsible for submitting SPV proofs and so on

Does it make sense